### DISCUSSION TITLE
If you want help - I have a Solax X3 Hybrid with 4x Solax batteries

### DISCUSSION THREAD

**Discussion by mickemartinsson** [wills106/homeassistant-solax-modbus#1] [General]
I have it connected with internal LAN and uses Pocket-Wifi for Solax-cloud.

**Comment by wills106** [comment]
Thanks, I am almost ready to release a new version with X3 support. So if you could feed back on which sensors need adjusting for scale or don't work at all etc.

**Reply by mickemartinsson** [reply]
Yes, of course!

**Reply by wills106** [reply]
Not sure what notifications you get. But I released version 0.2.0 earlier with X3 support.

It's still read only at the min.

**Comment by mickemartinsson** [comment]
Perfect. I'll start testing!!

**Comment by mickemartinsson** [comment]
### **Test report Solax homsassistant-solax-modbus Version 0.2.0**

**Installed equipment**
1 x SolaX X3 Hybrid
4 x Solax Tripple Power 5.7 kWh battery
1 x EPS switch
1 x Portable Wifi (Solax Cloud)
Lan connection (HA)
1 x Solax energy/feed meter
1 x Garo GLB car charger
1 x Garo Phase meter/load-balance meter

### Installing integration
Installing homsassistant-solax-modbus version 0.2.0
- No problems

Adding X3, extra sensors and EPS
- No problems

### Checking sensors with Solax logs

**SolaX Measured Power (neg = buying)** 
> - Working and correct value
> - Correct unit (power = W)

**SolaX Today's Yield**
> - Working and correct value
> - Correct unit (energy = kWh)

**SolaX Today's Export Energy**
> - Working and correct value (rounded to one decimal)
> - Correct unit (energy = kWh)

**SolaX Today's Import Energy**
> - Working and correct value (rounded to one decimal)
> - Correct unit (energy = kWh)

**SolaX house load**
> - Working and correct value
> - Correct unit (power = W)

**Solax voltage and current**
> - All values are the same as logs

**SolaX BMS Charge Max Current**
> - Unknown
> - No settings were found on the inverter that matches this

**SolaX BMS Discharge Max Current**
> - Unknown
> - No settings were found on the inverter that matches this

**Off-grid (EPS)**
> - Not tested yet

**SolaX Input Energy Charge Today**
> - Can’t verify this value
> - Not sure what it is

**SolaX Output Energy Charge Today**
> - Can’t verify this value
> - Not sure what it is

**SolaX Total Energy To Grid**
_sensor.solax_total_energy_to_grid_2_
> - Wrong value/wrong scale
> - Says: 1 367 343,1 kWh
> - My main meter says: 4 332,1 kWh (but that is two months older than Solax installation)

**"Energy" Dashboard**
> - No problem installing
> - Waiting for values and will verify that after 24h

**Battery SOC**
> - Missing

**Reply by wills106** [reply]
SolaX Today's Export Energy - Corrected to two decimal places
SolaX Today's Import Energy - Corrected to two decimal places

SolaX BMS Charge Max Current - Unknown on my Gen3 X1 as well
SolaX BMS Discharge Max Current - Unknown on my Gen3 X1 as well

SolaX Input Energy Charge Today - Only guessing, but I think this is the flow of power in and out of the battery / BMS?
SolaX Output Energy Charge Today - Only guessing, but I think this is the flow of power in and out of the battery / BMS?

SolaX Total Energy To Grid - Tweaked the values, but doesn't make sense too me?

Battery SOC - Should be sensor.solax_battery_capacity assuming you have named your setup the default solax

**Reply by mickemartinsson** [reply]
SolaX Input Energy Charge Today and SolaX Output Energy Charge Today - I agree, if I look at the flow from my batteries today that make sense. I'll calculate tomorrow as well.

I have removed the integration and installed it again - but no Battery SOC and no sensor solax_battery_capacity.

Total 90 sensors is installed from the integration.

Other values are working perfect! Great work!!

**Reply by wills106** [reply]
Do you not have a battery Icon when you click on SolaX in the integration page?

![image](https://user-images.githubusercontent.com/18155231/129485343-82f23dff-2708-4fcb-94d3-78bf9632d116.png)

**Reply by mickemartinsson** [reply]
Very strange. I have the battery icon and I have it on Devices but not on Lovelace/0 and can not use it on my other cards.
I renamed it on the Devices page - then I can use it on my on cards but still not in Lovelace/0...
Maybe it is because I have used Modbus from packages before.

**Reply by wills106** [reply]
Might be worth having a look in Developer Tools to see what the exact name of it is.

Perhaps you have a duplicate name somewhere and it's ended up sensor solax_battery_capacity_2 ?

Or can you click on the battery in the Integration page to find the name?

![image](https://user-images.githubusercontent.com/18155231/129485927-37ea4304-5267-4585-8882-bb1298b41524.png)

**Reply by mickemartinsson** [reply]
Yes, it is there and with the correct name. Couldn't find any duplicate name.
Did I a cold reboot - still not on Lovelace/0.
Tried to move it to another area - but still not on Lovelace/0
Tried other web (not cached) and IOS - still not on Lovelace/0
Can be used on other cards without problems.

**Reply by wills106** [reply]
All my Lovelace is manual setup. So not sure what's wrong with it?

**Reply by mickemartinsson** [reply]
It works on my own manual cards - so it's a minor problem but still strange. I use /0 just to see all data and names - it makes it easier when doing manual cards.

**Reply by MJP-76** [reply]
I have found some entities don't clean up, do you have and entities in your list that could have been named the same or are "restored"

**Reply by mickemartinsson** [reply]
Good point there. I did a look at all entities sorted on problems/status, names, and hidden. No duplicates or restored.
Also looked at automations if there were any old names there - found some but still no luck with /0.
Perhaps the old entities still are there somewhere in /0 and blocking the new one.

**Comment by mickemartinsson** [comment]
Working good! One problem with Energy - there are some strange values at midnight (0.00-01.00) where the solar-energy gets below 0 (see image below -9,3 kWh) so that all energy from solar panels during the day adds to the starting negative value. Is there a problem with daily reset in HA and daily reset in Solax?

**Reply by mickemartinsson** [reply]
![Skärmavbild 2021-08-17 kl  13 18 15](https://user-images.githubusercontent.com/4275111/129717162-c8f01ce2-4605-4093-9570-973670c06a57.png)

**Reply by wills106** [reply]
I think it's the daily total, that the dashboard generates. I have one at 23:00 It probably doesn't help that part of Home Assistant seems to be in UTC and the other half in GMT for me.

It's early days for the Energy Dashboard and not in real-time so I don't give it a great deal of attention at the moment.

![image](https://user-images.githubusercontent.com/18155231/129775145-b48bff14-5e1e-45f6-9778-35da558d6531.png)

**Reply by mickemartinsson** [reply]
It looks that it always calculates hour minus hour-1 (see post https://community.home-assistant.io/t/new-template-sensors-for-energy-start-new-day-with-negative-value/330651). So it should be a total without daily reset.

**Reply by mickemartinsson** [reply]
I made an addition to Utility meter without cycle period. Yield makes the battery integrated into production.

solax_total_energy:
&nbsp;  &nbsp;  source: sensor.solax_today_s_yield
solax_grid_energy_in:
&nbsp;   &nbsp;       source: sensor.solax_today_s_import_energy
solax_grid_energy_out:
&nbsp;   &nbsp;       source: sensor.solax_today_s_export_energy

**Reply by wills106** [reply]
They have changed the way STATE_CLASS_MEASUREMENT works by the looks of it.
I'll update the code and see if it changes the behaviour of the Energy Dashboard.

**Reply by wills106** [reply]
I have updated all of the sensors to the new style so hopefully everything should be ok now?

Check the update notes I have changed the naming of the battery power in/out, seems to work ok for me in the Energy Dashboard.

If you get chance it would be good if you could provide a Swedish translation file for this component?
I think a few members are using SolaX equipment in Sweden.

**Reply by mickemartinsson** [reply]
It works really well! Of course, I'll fix a Swedish translation!

**Reply by luisaraujoxx** [reply]
Hi Guys, Thank you for such nice wok Wills , i dont get tired of saying it.

How do you get the energy going to the baterry and comming out from the battery, instant, i used to use that sensor, but with the new release i just find, total day input and output. It's a X1 with 2x 5,8Kw/h 3P.
regards.

**Reply by wills106** [reply]
On 0.2.2 I renamed "Input Energy Charge Today" & "Output Energy Charge Today" to "Battery Input Energy Today" & "Battery Output Energy Today" You should be able to find them.

If you use the default hub name of SolaX it should be sensor.solax_battery_input_energy_today

Edit:
I miss read you sensor.solax_battery_power_charge
Positive - flow into battery
Negative - flow from battery

**Reply by luisaraujoxx** [reply]
i didnt missed the daily amount sensors, but i would like to access the instant energy in W for the battery usage along the day and check all energy coming in or out, on spot.

Used to be solax_battery_power.

**Reply by wills106** [reply]
I have edited the answer above

**Reply by luisaraujoxx** [reply]
sorry i missed this aldo :) sensor.solax_battery_power_charge

**Reply by wills106** [reply]
For a mini dashboard for Solar I am using power-distribution-card
Nice simple visualisation of real-time power flow.

![image](https://user-images.githubusercontent.com/18155231/137131695-ccfe8cc4-ef4a-4242-b741-cedabde70589.png)

**Reply by luisaraujoxx** [reply]
i will give a try, because the new energymonitor from hasso feels a litle stange:
![image](https://user-images.githubusercontent.com/74508869/137132580-aa62b633-6c43-4850-9cd5-30d6782b3fd3.png)

**Reply by wills106** [reply]
The built in one isn't real-time. It's always about an hour behind, even though the visuals look "real-time"

It's more for looking back in time compared to seeing what's going on at the moment.

**Comment by klio-klio** [comment]
somehow I don't quite understand what is needed for integration. I have installed it and do not get any values in. Although I can get the values to my PLC at home via Modbus TCP without any problems.

**Comment by klio-klio** [comment]
![Screenshot 2022-09-26 142330](https://user-images.githubusercontent.com/76972213/192275967-29db4574-6494-4e3c-b977-0abef335b928.jpg)
I also have X3 Hybrid G3 with battery

**Reply by wills106** [reply]
Do you have any errors in your log like "unrecognized inverter type - serial number : {your_serial_number_here}"
Have you disabled the built in Modbus connect with Home Assistant? The Inverter isn't happy having two connections to it.

**Comment by klio-klio** [comment]
There was no error message. do you have to enter my SN number? Homeassistant is only connected to my heat pump via Modbus. The inverter is permanently integrated into my PLC.

**Reply by wills106** [reply]
if everything is Unavailable I would expect to see some form of an error in the logs.

If you are using a Gen3 you should be able to just connect to the Inverter using the built in Ethernet connection. By passing any thrid party device.

What sort of PLC are you integrating with at the moment? Is it just a Waveshare type device?

**Comment by klio-klio** [comment]
now I have reset everything. and newly installed
![Screenshot 2022-09-26 155512](https://user-images.githubusercontent.com/76972213/192295367-0e98913e-45a8-4f0c-9e6e-d93ff5dcd4cc.jpg)
![Screenshot 2022-09-26 155032](https://user-images.githubusercontent.com/76972213/192295513-26373377-7ac9-4b97-8274-71029c05f158.jpg)

**Comment by infradom** [comment]
The deprecated attributes warning is not normal but is probably not the cause of the other errors.
Which version of our integration are you using? Looks like you are using an older version.
Is the modbus address (not only the IP address) correctly configured in HA ?

**Comment by klio-klio** [comment]
via HACS new version. installs.
![Screenshot 2022-09-26 224047](https://user-images.githubusercontent.com/76972213/192377404-58df1f74-8b36-4807-9a0a-ec3a513d4c2f.jpg)
Modbus Address: IP, port502, ID1

**Reply by wills106** [reply]
Is address 192.168.178.51 the address of the Built in Ethernet port of your Inverter or an address on this PLC you have setup?

If it's the address of you PLC setup. Is the registers mapped in exactly 1 to 1?
So is holding register 0x0 on the Inverter mapped to 0x0 on the PLC and 0x1 is mapped to 0x1 etc?

**Reply by infradom** [reply]
It might be possible that the Solax inverter only accepts one TCP/502 connection, not 2 (PLC and HA). So I wonder if it would work if the PLC is not connected.

**Reply by wills106** [reply]
Under Known Issues No1 it does mention you can't have multiple connections to the Inverter, when I wrote that statement it was in relation to Modbus over TCP.
Not sure if it's the case with RS485. Although you would soon get to a point where you would overwhelm the Inverter with too many requests.

**Reply by infradom** [reply]
In your drawing, it is very unclear which connections are Ethernet and which ones are serial RS485. I assume HA runs on the PC and the topmost horizontal line represents an Ethernet segment. It is unclear what the lower horizontal lines are physically

**Comment by klio-klio** [comment]
[scann.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/9654203/scann.pdf)
192.168.178.51 Inverter address

**Reply by infradom** [reply]
So if I understand it right, both HA (running on Lokaler PC) and Logo PLC connect to the inverter through tcp/ip
As @wills106 explained, multiple connections to the inverter may not work. I suggest to try first without the connected PLC

**Comment by klio-klio** [comment]
PLC disconnected . Everything worked. Thank you very much

**Comment by klio-klio** [comment]
My problem is not solved. how do I make everything work together? above all, what is the experience with the Hybrid G4? the has no internal network connection. only About Poket.

**Reply by wills106** [reply]
Do you need both the PLC and Home Assistant connected to your Inverter at the same time?
What exactly are you doing with the PLC connected to the Inverter?

The pocket WiFi does not provide you with a Modbus connection (well it does, if you use it in a very unsecure way)
If you want full control of your Gen4 with Home Assistant you will need to connect to it with either a RS485 - USB adaptor or a RS485 - Ethernet adaptor, ie a Waveshare.

**Reply by klio-klio** [reply]
with pocket LAN too?

**Reply by wills106** [reply]
Pocket LAN doesn't expose Modbus either.

**Comment by klio-klio** [comment]
I get a G4 inverter 15.0 as an upburst for my old G3 8.0

**Reply by infradom** [reply]
> I get a G4 inverter 15.0 as an upburst for my old G3 8.0

For G4, we really recommend the use of an RS485 to Ethernet adapter (as described in the Wiki). Such adapters might even be able to serve 2 clients (PLC and HA), but I have some doubts about that.
